blogs/data-mover/data-mover-overview.md

```diff
@@ -51,44 +51,88 @@ cluster. Due to this poor durability, in the case of a disaster scenario, the
-
+targeted backupStorageLocation (generally object storage), providing additional safety. 
+
+
+## Components
+
+### OADP Operator: 
+OADP is the OpenShift API for Data Protection operator. This open source operator sets up and installs Velero on the OpenShift platform, allowing users to backup and restore applications. 
+We will be installing Velero alongside the CSI plugin (modified version).
+
+### Modified CSI plugin (M-CSI):  
+The upstream Velero plugin for CSI is modified to facilitate CSI volumesnapshot data movement from OpenShift cluster to object storage and vice versa.
+
+### VolSync:
+VolSync is a Kubernetes operator that performs asynchronous replication of persistent volumes within, or across, clusters. The replication provided by VolSync is independent of the storage system. This allows replication to and from storage types that don’t normally support remote replication. 
+We will be using Volsync’s restic datamover.
+
+### VolumeSnapshotMover (VSM) Controller:
+The VSM controller is the CSI data movement orchestrator, it gets deployed via OADP Operator when you enable the datamover feature. This controller has the following responsibilities:
+- Validates the VolumeSnapshotBackup/VolumeSnapshotRestore Custom Resources.
+- Makes sure that the data movement workflow has the appropriate storage credentials
+- Performs the copy of VolumeSnapshotContent, CSI VolumeSnapshot and PersistentVolumeClaims from application namespace to OADP Operator namespace
+- Triggers the data movement process and subsequently performs the cleanup of extraneous resources created.
+
+### VolumeSnapshotMover CustomResourceDefinitions (CRDs):
+The data mover process will be based on 2 Custom Resource Definitions:
+- VolumeSnapshotBackup (VSB):
+```
+Spec:
+  Volumesnapshotcontent:
+  ProtectedNamespace:
+  ResticSecretRef:
+Status:
+  Completed:
+  SourcePVCData:
+  Conditions:
+  ResticRepository:
+  Phase:
+  VolumeSnapshotClassName:  
+```
+- VolumeSnapshotRestore (VSR):
+```
+Spec:
+  ResticSecretRef:
+  VolumeSnapshotMoverBackupRef:
+  ProtectedNamespace:
+Status:
+  Conditions:
+  Phase:
+  Snapshothandle:
+```
 
 ## Backup Process
 
-- (steps here)
+- The M-CSI plugin is extended to facilitate the data movement of CSI VolumeSnapshots(VS) from cluster to object storage.
+- When the Velero Backup is triggered, the M-CSI plugin creates a VS for each PersistentVolumeClaim (PVC) to be backed up.
+- Now for the created VS, the M-CSI plugin fetches the associated VolumeSnapshotContent (VSC) and adds it as an additional item to be backed up.
+- Subsequently, the M-CSI plugin then checks whether there is a VolumeSnapshotBackup (VSB) instance associated with the VSC that was added as an additional item, if there isn't one then the M-CSI plugin creates a VSB for each VSC.
+- The creation of VSB triggers the data movement process, the VolumeSnapshotMover (VSM) controller reconciles on the VSB instance and begins the snapshot movement process.
```

Comment:
This sentence is a bit confusing, as it mentions both data movement and snapshot movement. What about something like "The creation of a VSB triggers the data movement process as the VolumeSnapshotMover (VSM) controller begins to reconcile on this VSB instance."